radicle-node/src/service.rs

```diff
@@ -884,8 +884,9 @@ where
                     }
                 }
             }
-            (session::State::Connected { .. }, Message::Upgrade { .. }) => {
-                todo!();
+            (session::State::Connected { .. }, Message::Upgrade { repo }) => {
+                // All we need is to instruct the transport to handover to the worker
+                self.reactor.fetch(*remote, repo, Namespaces::default());
```

Comment:
We're going to need the direction of the fetch also right? Think I forgot to add that.